arcgis-ios-sdk-samples/Utility network/Display content of utility network container/DisplayContentUtilityNetworkContainerViewController.swift

```diff
@@ -0,0 +1,360 @@
+    @IBAction func exitContainerView() {
+        // Disable the bar button item since container view will be exited.
+        exitBarButtonItem.isEnabled = false
+        // Remove all the objects that were added onto the graphics overlay.
+        graphicsOverlay.graphics.removeAllObjects()
+        (mapView.map?.operationalLayers as? [AGSLayer])?.forEach { layer in
+            // Make each operational layer visible.
+            layer.isVisible = true
+        }
+        if let previousViewpoint = previousViewpoint {
+            // Return to the viewpoint before container view was entered.
+            mapView.setViewpoint(previousViewpoint) { [weak self] _ in
+                // Enable interaction on the map view.
+                self?.mapView.isUserInteractionEnabled = true
+            }
+        }
+    }
+    
+    /// Create a map.
+    ///
+    /// - Returns: An `AGSMap` object.
+    func makeMap() {
+        let webMapURL = URL(string: "https://sampleserver7.arcgisonline.com/portal/home/item.html?id=813eda749a9444e4a9d833a4db19e1c8")!
+        let map = AGSMap(url: webMapURL)
+        mapView.map = map
+    }
+    
+    /// Create and load the utility network using the feature service URL.
+    func loadUtilityNetwork() {
+        guard let map = mapView.map else { return }
+        utilityNetwork = AGSUtilityNetwork(url: featureServiceURL, map: map)
+        utilityNetwork?.load { [weak self] error in
+            if let error = error {
+                self?.presentAlert(error: error)
+            } else {
+                // Add self as the touch delegate for the map view.
+                self?.mapView.touchDelegate = self
+            }
+        }
+    }
+    
+    /// Get the legend information provided by the feature layers used in the utility network.
+    func fetchLegendInfo() {
+        // Create feature tables from URLs.
+        let electricDistributionTable = AGSServiceFeatureTable(url: featureServiceURL.appendingPathComponent("1"))
+        let structureJunctionTable = AGSServiceFeatureTable(url: featureServiceURL.appendingPathComponent("5"))
+        let legendGroup = DispatchGroup()
+        // Create feature layers using the feature tables.
+        featureLayers = [electricDistributionTable, structureJunctionTable].map(AGSFeatureLayer.init)
+        featureLayers.forEach { layer in
+            legendGroup.enter()
+            // Get the legend information of each layer.
+            layer.fetchLegendInfos { [weak self] legendInfos, _ in
+                defer { legendGroup.leave() }
+                guard let legendInfos = legendInfos else { return }
+                self?.legendInfos.append(contentsOf: legendInfos)
+            }
+        }
+        legendGroup.notify(queue: .main) { [weak self] in
+            guard let self = self else { return }
+            let featureLayerSymbols: [(String, AGSSymbol)] = self.legendInfos.compactMap { legendInfo in
```

Comment:
I gave it a shot but it seems like there are some impediments that make doing so not worth it. Each feature layer contains an `AGSLegendInfo` with the name "Unknown". Having duplicate keys causes an error in making the dictionary. I realize having duplicate `AGSLegendInfo`s also isn't ideal but there doesn't seem to be an easy way to remove just one duplicate without hardcoding the string "Unknown". I was okay with having 2 "Unknown"s since they technically come from different legends. What're your thoughts?

Reply:
How are you creating the dictionary?

Reply:
```swift
let featureLayerSymbols = [String: AGSSymbol](uniqueKeysWithValues: accumulatedLegendInfos.compactMap { legendInfo in
     if !legendInfo.name.isEmpty,
        let symbol = legendInfo.symbol {
           return (legendInfo.name, symbol)
         } else {
             return nil
          }
})
```
This is how I tried to create the dictionary. I got an error that said "Duplicate keys" due to having 2 `AGSLegendInfo`s with the name "Unknown" populated from each feature layer.

Reply:
Could you use [`Dictionary.init(_:uniquingKeysWith:)`](https://developer.apple.com/documentation/swift/dictionary/3127161-init) instead?

Reply:
``` swift
let symbols: [(String, AGSSymbol)] = accumulatedLegendInfos.compactMap { legendInfo in
                if !legendInfo.name.isEmpty,
                   let symbol = legendInfo.symbol {
                    return (legendInfo.name, symbol)
                } else {
                    return nil
                }
            }
            let featureLayerSymbols = Dictionary(symbols, uniquingKeysWith:  { (first, _) in first})
```
I could use `Dictionary.init(_:uniquingKeysWith:)` but it would still require making an array of tuples first.